### PR TITLE
docgen: Omit unknown type tag from Markdown format output

### PR DESCRIPTION
**Pull request by aduth**
Related: #18045
Unblocks: #19520

This pull request seeks to resolve an issue where, if `docgen` encounters a type that it cannot parse, the Markdown formatter will output a `null` literal value for the type. With these changes, it will ~simply not output a type at all~ output text indicating that the type is unknown, in order to avoid misleading the reader into thinking the type is `null` when it is not. 

This should be considered an interim solution to the larger problem of substituting Doctrine (#18045) in order to use more complex TypeScript types (#18838).

**Testing Instructions:**

Verify unit tests pass:

```
npm run test-unit
```

Verify there are no changed documentation after running:

```
npm run docs:build
```

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
It looks like a reasonable way to handle such cases for the time being 👍

File: packages/docgen/src/markdown/formatter.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ module.exports = function( rootDir, docPath, symbols, headingTitle, headingStart
 			formatTag(
 				'Parameters',
 				getSymbolTagsByName( symbol, 'param' ),
-				( tag ) => `\n- *${ tag.name }* \`${ tag.type }\`: ${ cleanSpaces( tag.description ) }`,
+				( tag ) => `\n- *${ tag.name }*${ tag.type ? ` \`${ tag.type }\`` : '' }: ${ cleanSpaces( tag.description ) }`,
```
Comment (nosolosw) [inline packages/docgen/src/markdown/formatter.js]: How about explicitly state that the type is `unknown`? I think it provides more clarity to the reader of the API than nothing at all.

Comment (aduth) [reply packages/docgen/src/markdown/formatter.js]: It's possible, yeah. I wasn't sure how best to call it in a way where it wouldn't be misinterpreted as the type (even "Unknown" could mislead one into reading it as some form of type). I'm on the fence whether it's more clear one way than the other, especially in mind of this.

Comment (nosolosw) [reply packages/docgen/src/markdown/formatter.js]: I'm not attached particularly to `unknown`, perhaps `UnknownType` is a better choice? Whatever works is fine for me. My point was more about this being like any other situation: we want to gracefully degrade and communicate the user what's happening.

Comment (aduth) [reply packages/docgen/src/markdown/formatter.js]: I added "(unknown type)".

I'm hoping it's only needed as an interim solution anyways, 'til we can improve types parsing here.

**Comment by aduth** [comment]
I somehow missed that types are output for a number of tags, so I updated in e9da943 to output the "unknown type" more consistently (caught a few more cases in the subsequent 4e10bce).

**Pull request merged**